oauth2_http/java/com/google/auth/oauth2/S2A.java

```diff
@@ -0,0 +1,110 @@
+package com.google.auth.oauth2;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.json.JsonObjectParser;
+import com.google.api.client.util.GenericData;
+import com.google.auth.http.HttpTransportFactory;
+import com.google.common.collect.Iterables;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ServiceLoader;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Utilities to fetch the S2A (Secure Session Agent) address from the mTLS configuration.
+ *
+ * <p>mTLS configuration is queried from the MDS MTLS Autoconfiguration endpoint.
+ */
+@ThreadSafe
+public final class S2A {
+  public static final String S2A_CONFIG_ENDPOINT_POSTFIX =
+      "/computeMetadata/v1/instance/platform-security/auto-mtls-configuration";
+
+  public static final String METADATA_FLAVOR = "Metadata-Flavor";
+  public static final String GOOGLE = "Google";
+  private static final int MAX_MDS_PING_TRIES = 3;
```

Comment:
Can we use the default retry value here: https://github.com/googleapis/google-auth-library-java/blob/d4a3763799896a35e27b9f5c178f77364e41668d/oauth2_http/java/com/google/auth/oauth2/OAuth2Utils.java#L110?

Reply:
Done in https://github.com/googleapis/google-auth-library-java/pull/1400/commits/c3ede1d32b1b4c3cd4c336f9c88a0eba399634e5